File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int len = 0;
        ListNode curr = head;

        while(curr != null) {
            len++;
            curr = curr.next;
        }

        int target = len - n;
        if(target == 0) return head.next;
        curr = head;
        for(int i=0; i<target-1; i++) {
            curr = curr.next;
        }
        curr.next = curr.next.next;
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        curr = head
        length = 0

        while curr is not None:
            length += 1
            curr = curr.next

        if length == n:
            return head.next

        curr = head
        nLength = length - n
        for i in range(nLength-1) :
            curr = curr.next

        curr.next = curr.next.next
        return head
        